packages/duckdb-wasm/src/bindings/bindings_base.ts

```diff
@@ -222,13 +222,19 @@ export abstract class DuckDBBindingsBase implements DuckDBBindings {
         return this.mod.ccall('duckdb_web_pending_query_cancel', 'boolean', ['number'], [conn]);
     }
     /** Fetch query results */
-    public fetchQueryResults(conn: number): Uint8Array {
+    public fetchQueryResults(conn: number): Uint8Array | null {
         const [s, d, n] = callSRet(this.mod, 'duckdb_web_query_fetch_results', ['number'], [conn]);
         if (s !== StatusCode.SUCCESS) {
             throw new Error(readString(this.mod, d, n));
         }
+
         const res = copyBuffer(this.mod, d, n);
         dropResponseBuffers(this.mod);
```

Comment:
```suggestion
        dropResponseBuffers(this.mod);
```
I am not sure if this needs to be performed, or moved afterwards.

Reply:
Yeah it may need to be revisited though might be worth a dedicated PR?

Reply:
I was thinking moving this below the `return null` might be it, but I need to check.
In any case, this impact only `retry` cases, that are currently not exposed in the wild, so might be ok-ish.

Reply:
I think we want it in both "SUCCESS" cases since a buffer is instantiated, otherwise we would leak the (1-byte) retry buffer each time no?